src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,94 @@
+package duke.parser;
+
+import duke.command.*;
+import duke.model.Deadline;
+import duke.model.Event;
+import duke.model.ToDo;
+
+import static java.lang.Integer.parseInt;
+
+/**
+ * Deals with making sense of the user command.
+ */
+public class Parser {
+
+    enum Keyword {
+        BYE("bye"),
+        LIST("list"),
+        MARK("mark"),
+        UNMARK("unmark"),
+        TODO("todo"),
+        DEADLINE("deadline"),
+        EVENT("event"),
+        DELETE("delete"),
+        FIND("find");
+
+        private String value;
+
+        /**
+         * A constructor for a Keyword.
+         *
+         * @param keyword a string indicating the user's intentions
+         */
+        Keyword(String keyword) {
+            this.value = keyword;
+        }
+
+        /**
+         * Finds and returns a Keyword based on the user's input Keyword.
+         *
+         * @param inputKeyword the user's input Keyword
+         * @return the corresponding Keyword related to a Command
+         */
+        public static Keyword getKeyword(String inputKeyword) {
+            for (Keyword k : Keyword.values()) {
+                if (inputKeyword.toLowerCase().equals(k.value)) {
+                    return k;
+                }
+            }
+            return null;
+        }
+    }
+
+    /**
+     * Returns the corresponding Command with respect to the user's input.
+     *
+     * @param userInput the user's input
+     * @return a Command associated with the user's input
+     */
+    public static Command parse(String userInput) {
+        String[] input = userInput.split(" ", 2);
+
+        String inputKeyword = input[0];
+
+        Keyword keyword = Keyword.getKeyword(inputKeyword);
+
+        String[] description;
+
+        switch (keyword) {
+            case BYE:
```

Comment:
There should not be any indentation for the cases clauses, so the word 'case' should have the same indentation as switch